### PR TITLE
make SummaryConfig constructible from variables

### PR DESCRIPTION
**Pull request by akva2**
also add accessors and equality operator

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-simulators=2268 please

File: src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp [hunk 1]
```diff
@@ -795,6 +795,23 @@ inline void handleKW( SummaryConfig::keyword_list& list,
...
+    : keyword_(kword)
+    , category_(cat)
+    , loc(locat)
+    , type_(type)
+    , name_(name)
+    , number_(number)
+    , userDefined_(udef)
+{
+}
+
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp]: Technically you could achieve the same effect using the functionality already in place:
```C++
auto snode = SummaryNode{ kword, cat, locat }
    .parameterType(type)
    .namedEntity(name)
    .number(number)
    .isUserDefined(udef);
```

Comment (akva2) [reply src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp]: aha, i did not notice the setters (usually not there). sure, i can use those instead if you prefer.

**Comment by akva2** [comment]
jenkins build this opm-simulators=2268 please

**Pull request merged**